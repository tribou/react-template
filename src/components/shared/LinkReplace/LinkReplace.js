// @flow
import React, { PureComponent } from 'react'

type ReactProps = {
  location: Object,
  history: Object,
  match: Object,
  staticContext: ?Object,
  to: Object,
  className: any,
  children: React$Element<any> | string,
}

type Props = ReactProps

class LinkReplace extends PureComponent<Props> {

  // TODO: watch for resolution of
  // https://github.com/yannickcr/eslint-plugin-react/issues/1376
  props: Props

  handleClick = () => {

    const {
      history,
      to,
    } = this.props

    if (to.state && to.query) {

      history.replace({
        pathname: to.pathname,
        state: to.state,
        query: to.query,
      })

    }

    else if (to.state) {

      history.replace({
        pathname: to.pathname,
        state: to.state,
      })

    }

    else if (to.query) {

      history.replace({
        pathname: to.pathname,
        query: to.query,
      })

    }

    else {

      history.replace({
        pathname: to,
        state: { modal: '' },
      })

    }

  }

  render () {

    return (
      <div
        role="button"
        tabIndex={-1}
        className={this.props.className}
        onClick={this.handleClick}
      >
        {this.props.children}
      </div>
    )


  }

}

export default LinkReplace
