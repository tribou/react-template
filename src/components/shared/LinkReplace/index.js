// @flow
import { withRouter } from 'react-router'
import LinkReplace from './LinkReplace'

export default withRouter(LinkReplace)
