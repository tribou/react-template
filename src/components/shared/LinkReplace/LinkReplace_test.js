// @flow
/* eslint-disable max-len */

// Enzyme docs:
// http://airbnb.io/enzyme/docs/api/index.html

import React from 'react'
import { shallow } from 'enzyme'
import LinkReplace from './LinkReplace'

const mockProps = {
  className: 'overriding-class',
  children: 'Link Text',
}

it('<LinkReplace> renders the LinkReplace component', () => {

  const wrapper = shallow(
    <LinkReplace
      {...mockProps}
      to={{
        pathname: 'location_string',
        state: { modal: 'create-offer-intro' },
      }}
    >
      <button />
    </LinkReplace>
  )

  expect(wrapper).toMatchSnapshot()

})
