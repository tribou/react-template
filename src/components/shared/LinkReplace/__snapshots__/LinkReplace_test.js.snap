// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkReplace> renders the LinkReplace component 1`] = `
<div
  className="overriding-class"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={-1}
>
  <button />
</div>
`;
